test/unit/srv_option_handling.test.ts

```diff
@@ -0,0 +1,119 @@
+import { expect } from 'chai';
+import * as dns from 'dns';
+import * as sinon from 'sinon';
+import { promisify } from 'util';
+
+import { MongoCredentials } from '../../src/cmap/auth/mongo_credentials';
+import { $EXTERNAL_AUTH_SOURCE_MECHANISMS, AuthMechanism } from '../../src/cmap/auth/providers';
+import { resolveSRVRecord } from '../../src/connection_string';
+
+const resolveSRVRecordAsync = promisify(resolveSRVRecord);
+
+describe('Srv Option Handling', () => {
+  let resolveSrvStub: sinon.SinonStub;
+  let resolveTxtStub: sinon.SinonStub;
+  let resolveSRVRecordStub: sinon.SinonStub;
+  let lookupStub: sinon.SinonStub;
+
+  afterEach(async () => {
+    if (resolveSrvStub) {
+      resolveSrvStub.restore();
+      resolveSrvStub = undefined;
+    }
+    if (resolveTxtStub) {
+      resolveTxtStub.restore();
+      resolveTxtStub = undefined;
+    }
+    if (lookupStub) {
+      lookupStub.restore();
+      lookupStub = undefined;
+    }
+    if (resolveSRVRecordStub) {
+      resolveSRVRecordStub.restore();
+      resolveSRVRecordStub = undefined;
+    }
+  });
+
+  function makeStub(txtRecord: string) {
+    const mockAddress = [
+      {
+        name: 'localhost.test.mock.test.build.10gen.cc',
+        port: 2017,
+        weight: 0,
+        priority: 0
+      }
+    ];
+
+    const mockRecord: string[][] = [[txtRecord]];
+
+    // first call is for the driver initial connection
+    // second call will check the poller
+    resolveSrvStub = sinon.stub(dns, 'resolveSrv').callsFake((address, callback) => {
+      return process.nextTick(callback, null, mockAddress);
+    });
+
+    resolveTxtStub = sinon.stub(dns, 'resolveTxt').callsFake((address, thisIsWhatWeAreTesting) => {
+      thisIsWhatWeAreTesting(null, mockRecord);
+    });
+  }
+
+  for (const mechanism of $EXTERNAL_AUTH_SOURCE_MECHANISMS) {
+    it('should set authSource to $external for ${mechanism} external mechanism:', async function () {
```

Comment:
I think you meant this to be a template string

Reply:
yep, good catch!